src/lower/mod.rs

```diff
@@ -898,16 +939,73 @@ impl<'r> Lowerer<'r> {
         }
     }
 
+    fn ip_memcmp(&mut self, left: Operand, right: Operand) -> Var {
+        // For an ip addr, we need to be clever, because we can
+        // only compare the bytes that are used the current
+        // variant. Therefore, we read the discriminant of the
+        // left value. If that's 0, we compare 5 bytes,
+        // otherwise, we compare 17 bytes. In other words we
+        // compare 5 + 12 * discriminant bytes.
```

Comment:
This is interesting, but I think a branch condition would be simpler and faster.

Reply:
Not simpler, because branching is a bit hard to add at this level (with the current API). Also keep in mind that cranelift optimizes branches very poorly at the moment, that being said, faster is not even on my mind right now.